### PR TITLE
ceph-volume: fix Batch object in py3 environments

### PR DESCRIPTION
**Pull request by jan--f**
Also add a smoketest for the Batch object.

Fixes: https://tracker.ceph.com/issues/37356

Signed-off-by: Jan Fajerski <jfajerski@suse.com>

### CODE REVIEW

File: src/ceph-volume/ceph_volume/devices/lvm/batch.py [hunk 1]
```diff
@@ -139,14 +128,21 @@ def __init__(self, argv):
...
-            if detail.get('partitions') != {}:
-                del all_devices[device]
-        devices = sorted(all_devices.items(), key=lambda x: (x[0], x[1]['size']))
-        return device_formatter(devices)
+        devices = [(device, details) for device, details in
+                       disk.get_devices().items() if details.get('partitions') == {}]
+        size_sort = lambda x: x[1]['size']
+        return self._device_formatter(sorted(devices, key=size_sort))
+
+    def _device_formatter(self, devices):
```
Comment (alfredodeza) [inline src/ceph-volume/ceph_volume/devices/lvm/batch.py]: why is there a need to move this into a method?

**Comment by alfredodeza** [comment]
jenkins test ceph-volume tox

File: src/ceph-volume/ceph_volume/devices/lvm/batch.py (hunk 1 above)
Comment (jan--f) [reply src/ceph-volume/ceph_volume/devices/lvm/batch.py]: It's only used in the Batch object. No need for a module global function.

Comment (jan--f) [reply src/ceph-volume/ceph_volume/devices/lvm/batch.py]: Its also quite tightly coupled to the output of ```get_devices()```

Comment (alfredodeza) [reply src/ceph-volume/ceph_volume/devices/lvm/batch.py]: I would say that if it doesn't need to be moved lets leave it? Lets focus on the bug only

Comment (jan--f) [reply src/ceph-volume/ceph_volume/devices/lvm/batch.py]: What's your argument for have the formatting function in the global scope?

**Comment by jan--f** [comment]
retest this please

**Comment by alfredodeza** [comment]
@jan--f if you could leave out the changes that are not explicit to the bug (moving the two functions into the class) I think we can get this merged. Thanks!

**Pull request merged**